x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBytes.java

```diff
@@ -256,7 +296,15 @@ public int write(ByteBuffer src, long position) throws IOException {
             assert position % PAGE_SIZE == 0;
             assert src.remaining() % PAGE_SIZE == 0;
             checkOffsets(position, src.remaining());
-            final int bytesWritten = fileChannel.write(src, position);
+            final int bytesWritten;
+            if (mmap) {
+                bytesWritten = src.remaining();
+                mappedByteBuffer.put(Math.toIntExact(position - pageStart), src, src.position(), bytesWritten);
+                mappedByteBuffer.force();
```

Comment:
Looking at the javadoc of the method:
> Forces any changes made to this buffer's content to be written to the storage device containing the mapped file.

I am wondering if we really need to force that?

Reply:
:facepalm:  ofc not that was just from debugging yesterday, thanks for spotting this would've ruined all our benchmarking :)